Improve the API documentation and markdown files (#177)
ballerina-platform/module-ballerinax-rabbitmq: rabbitmq-ballerina/Module.md

## Overview

This module provides the capability to send and receive messages by connecting to the RabbitMQ server.

> RabbitMQ gives your applications a common platform to send and receive messages, and your messages a safe place to live until received. RabbitMQ is one of the most popular open source message brokers. It is lightweight and easy to deploy on premises and in the cloud.

Reviewer: ```suggestion
RabbitMQ gives your applications a common platform to send and receive messages and a safe place for your messages to live until received. RabbitMQ is one of the most popular open-source message brokers. It is lightweight and easy to deploy on-premise and in the cloud.
```

### Basic Usage

#### Setting Up the Connection

First, you need to set up the connection with the RabbitMQ server. The following ways can be used to connect to a
RabbitMQ server.

1. Connect to a RabbitMQ node with the default host and port:
```ballerina
    rabbitmq:Client rabbitmqClient = check new(rabbitmq:DEFAULT_HOST, rabbitmq:DEFAULT_PORT);
```

2. Connect to a RabbitMQ node with a custom host and port:
```ballerina
    rabbitmq:Client rabbitmqClient = check new("localhost", 5672);
```

3. Connect to a RabbitMQ node with host, port, and additional configurations:
```ballerina
    rabbitmq:ConnectionConfiguration config = {
        username: "ballerina",
        password: "password"
    };
    rabbitmq:Client rabbitmqClient = check new("localhost", 5672, configs);
```

The `rabbitmq:Client` can now be used to send and receive messages as described in the subsequent sections.

#### Using Exchanges and Queues

> Client applications work with exchanges and queues, which are the high-level building blocks of the AMQP protocol. These must be declared before they can be used. The following code declares an exchange, and a server-named queue and then binds them together.

Reviewer: ```suggestion
Client applications work with exchanges and queues, which are the high-level building blocks of the AMQP protocol. These must be declared before they can be used. The following code declares an exchange and a server-named queue and then binds them together.
```

```ballerina
    check rabbitmqClient->exchangeDeclare("MyExchange", rabbitmq:DIRECT_EXCHANGE);
    check rabbitmqClient->queueDeclare("MyQueue");
    check rabbitmqClient->queueBind("MyQueue", "MyExchange", "routing-key");
```

This sample code will declare,
- a durable auto-delete exchange of the type `rabbitmq:DIRECT_EXCHANGE`
- a non-durable, exclusive auto-delete queue with an auto-generated name

Next, the `queueBind` function is called to bind the queue to the exchange with the given routing key.

```ballerina
    check rabbitmqClient->exchangeDeclare("MyExchange", rabbitmq:DIRECT_EXCHANGE);
    check rabbitmqClient->queueDeclare("MyQueue", { durable: true,
                                                   exclusive: false,
                                                   autoDelete: false });
    check rabbitmqClient->queueBind("MyQueue", "MyExchange", "routing-key");
```

This sample code will declare,
- a durable auto-delete exchange of the type `rabbitmq:DIRECT_EXCHANGE`

> - a durable, non-exclusive non-auto-delete queue with a well-known name

Reviewer: ```suggestion
- a durable, non-exclusive, non-auto-delete queue with a well-known name
```

1. Listen to incoming messages with the `onMessage` remote method:

```ballerina
    listener rabbitmq:Listener channelListener= new(rabbitmq:DEFAULT_HOST, rabbitmq:DEFAULT_PORT);
    
    @rabbitmq:ServiceConfig {
        queueName: "MyQueue"
    }
    service rabbitmq:Service on channelListener {
        remote function onMessage(rabbitmq:Message message) {
        }
    }
```

2. Listen to incoming messages and reply directly with the `onRequest` remote method:

```ballerina
    listener rabbitmq:Listener channelListener= new(rabbitmq:DEFAULT_HOST, rabbitmq:DEFAULT_PORT);
    
    @rabbitmq:ServiceConfig {
        queueName: "MyQueue"
    }
    service rabbitmq:Service on channelListener {
        remote function onRequest(rabbitmq:Message message) returns string {
            return "Hello Back!";
        }
    }
```

The `rabbitmq:Message` record received can be used to retrieve its contents.

### Advanced Usage

#### Client Acknowledgements

The message consuming is supported by mainly two types of acknowledgement modes, which are auto acknowledgements and client acknowledgements.

> Client acknowledgements can further be divided into to two different types as positive and negative acknowledgements.

Reviewer: ```suggestion
Client acknowledgements can further be divided into two different types as positive and negative acknowledgements.
```

> The default acknowledgement mode is auto-ack (messages are acknowledged immediately after consuming). Following examples show the usage of positive and negative acknowledgements.

Reviewer: ```suggestion
The default acknowledgement mode is auto-ack (messages are acknowledged immediately after consuming). The nfollowing examples show the usage of positive and negative acknowledgements.
```

Author: This suggestion has a type

Reviewer: Corrected. Thanks!

> WARNING: To ensure the reliability of receiving messages, use the client-ack mode.

1. Positive client acknowledgement:
```ballerina
    listener rabbitmq:Listener channelListener= new(rabbitmq:DEFAULT_HOST, rabbitmq:DEFAULT_PORT);
    
    @rabbitmq:ServiceConfig {
        queueName: "MyQueue",
        autoAck: false
    }
    service rabbitmq:Service on channelListener {
        remote function onMessage(rabbitmq:Message message, rabbitmq:Caller caller) {
            rabbitmq:Error? result = caller->basicAck();
        }
    }
```

2. Negative client acknowledgement:
```ballerina
    listener rabbitmq:Listener channelListener= new(rabbitmq:DEFAULT_HOST, rabbitmq:DEFAULT_PORT);
    
    @rabbitmq:ServiceConfig {
        queueName: "MyQueue",
        autoAck: false
    }
    service rabbitmq:Service on channelListener {
        remote function onMessage(rabbitmq:Message message) {
            rabbitmq:Error? result = caller->basicNack(true, requeue = false);
        }
    }
```

> The negatively-acknowledged (rejected) messages can be re-queued byt setting `requeue` to `true`.

Reviewer: ```suggestion
The negatively-acknowledged (rejected) messages can be re-queued by setting the `requeue` to `true`.
```
